File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
object Raindrops {

    fun convert(n: Int): String {
        var rain : String = ""

        for (i in 1..8) {
            if (n % i == 0) {
                if (i == 3) {
                    rain += "Pling"
                } else if (i == 5) {
                    rain += "Plang"
                } else if (i == 7) {
                    rain += "Plong"
                }
            }
        }

        return if (rain == "") n.toString() else rain
    }
}

File: go/raindrops/raindrops.go
// Package raindrops contains a Convert function.
package raindrops

import "strconv"

// Convert takes an integer and returns the rain drop sound string, or the integer.
func Convert(n int) string {
	rain := ""

	if n%3 == 0 {
		rain += "Pling"
	}
	if n%5 == 0 {
		rain += "Plang"
	}
	if n%7 == 0 {
		rain += "Plong"
	}

	if rain == "" {
		return strconv.Itoa(n)
	}

	return rain
}
